### ISSUE TITLE
[otp_ctrl] DAI rd/wr access between digest and reset

### ISSUE THREAD

**Issue by cindychip** [lowRISC/opentitan#4814]
Hi Michael,

I thought we talked about this but could not find any email/issues related to it. So I filed this one for tracking. Please feel free to mark it as duplicated if you find the original thread.

The issue comes as the sequence below:
1. lock partition (say CreatorSwPartition) with CSR write
2. Issue Partition CreatorSw read/write via DAI interface
3. Issue reset

During sequence 2), current RTL allows read/write to happen without any error. Do you plan to add some error report for sequence 2? Shall we block the write access?

Thanks,
Cindy

**Comment by msfschaffner** [comment]
Yes, we have already discussed this I think.

The behavior you describe is due to how the OTP is structured internally.
I.e. the read/write access locks are determined based on the digest values that are read upon initialization of the OTP controller.
Hence, in order to take effect, a reset is required.

I currently don't plan to change this behavior since it would make the internal structure more complex and less secure (i.e., the DAI would need additional connections to the partition controllers to change values inside the buffer registers).

Hence, I have added some additional guidance to the programmers model that highlights this behavior here:
https://docs.opentitan.org/hw/ip/otp_ctrl/doc/index.html#locking-partitions

Does that make sense?
